frontend/src/components/App.js

```diff
@@ -10,13 +10,13 @@ const App = () => {
   useEffect(() => {
     fetch(`data/${window.django.course_id}`)
       .then((response) => {
-        if (response.status > 400) {
-          return dispatch({ type: 'receivePayloadError' });
+        if (response.status >= 400) {
+          return dispatch({ type: 'businessDataFetchErrored', value: 'The canvas api fetch failed.' });
```

Comment:
It may be overkill, but one pattern I've seen is to use constants for event types, so that static checking can help with typos.

Reply:
That's a good idea

Reply:
In addition to the static checking for typos, it can also be helpful to have the constants declared in one place so you can see all of the available event types.

Reply:
Yeah, I think it's a good idea to employ constants. But there is a place in `Selector.js` where I use a template string to build the name of the event type, based on the props of the selector. Something like:
```js
let selectorName = 'showingRubrics'; // this comes from props
dispatch({ type: `${selectorName}Selected`, value: event.target.innerText });

// creates event called "showingRubricsSelected"
```
I suppose I should probably refactor that into a single event type, and pass the key of the selector as a parameter. Something like:
```js
import { selectorSelected } from './placeWhereConstantsAre';
let selectorName = 'showingRubrics'; // this comes from props
dispatch({ type: selectorSelected, selectorKey: selectorName, value: event.target.innerText });

// then handle with care in the reducer
```